### PR TITLE
Convert LinkExpirationFormatterMethod to Java's ChoiceFormat pattern

### LINKED ISSUE #21887: Simplify translations for LinkExpirationFormatterMethod
### Description

Currently the LinkExpirationFormatterMethod uses one property appended with a number to differentiate for example "second" and seconds" depending on the count of seconds.

This makes it difficult to use computer aided translation tools (CAT). This issue will transform this to a standard Java [ChoiceFormat](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/text/ChoiceFormat.html). 

### Discussion

_No response_

### Motivation

Support CAT tools like Weblate

### Details

I'm about to create a PR.

### PR DESCRIPTION
**Pull request by ahus1**
Closes #21887

The code that was used to convert the different languages (except English) is present in `LinkExpirationFormatterMethodTest#convert`. 

To test the result, consider for example `jshell`: 

``` 
jshell> java.text.MessageFormat.format("{0,choice,0#Sekunden|1#Sekunde|1<Sekunden}",2)
==> "Sekunden"


jshell> java.text.MessageFormat.format("{0,choice,0#Sekunden|1#Sekunde|1<Sekunden}",1)
==> "Sekunde"
```

Discussion: https://github.com/keycloak/keycloak/discussions/9270

### CODE REVIEW

**Comment by ahus1** [comment]
@hmlnarik - as we've talked about translation formats recently, I wonder if you could review and merge this PR as well. I wasn't sure where to put the code for the conversion, so I placed it in a test case as it can be run interactively from the IDE. I could also remove it from there and paste it into this PR. WDYT?

Thanks!

File: themes/src/main/resources/theme/base/email/messages/messages_en.properties [hunk 1]
```diff
@@ -39,18 +39,11 @@ requiredAction.VERIFY_EMAIL=Verify Email
...
-linkExpirationFormatter.timePeriodUnit.minutes.1=minute
-#for language which have more unit plural forms depending on the value (eg. Czech and other Slavic langs) you can override unit text for some other values like this: 
-#linkExpirationFormatter.timePeriodUnit.minutes.2=minuty
-#linkExpirationFormatter.timePeriodUnit.minutes.3=minuty
-#linkExpirationFormatter.timePeriodUnit.minutes.4=minuty
-linkExpirationFormatter.timePeriodUnit.hours=hours
-linkExpirationFormatter.timePeriodUnit.hours.1=hour
-linkExpirationFormatter.timePeriodUnit.days=days
-linkExpirationFormatter.timePeriodUnit.days.1=day
+linkExpirationFormatter.timePeriodUnit.seconds={0,choice,0#seconds|1#second|1<seconds}
```
Comment (stianst) [inline themes/src/main/resources/theme/base/email/messages/messages_en.properties:42]: ChoiceFormat is news to me ;)

Can't this be simplified to:
```suggestion
linkExpirationFormatter.timePeriodUnit.seconds=0#seconds|1#second|1<seconds
```

Comment (stianst) [reply themes/src/main/resources/theme/base/email/messages/messages_en.properties:42]: At least that's the format I get from trying to understand https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/text/ChoiceFormat.html

Comment (ahus1) [reply themes/src/main/resources/theme/base/email/messages/messages_en.properties:42]: IMHO the translation tools would recognize a Java message format `{0,choice,...}` and then validate it accordingly. So it is the best format for translation.

For some of the languages which didn't have any plurals (like `ja` and `tr` locales), where this is then very simple, although this is the exception.

So I'd like to proceed with this setup.

Comment (stianst) [reply themes/src/main/resources/theme/base/email/messages/messages_en.properties:42]: Looking at https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/text/MessageFormat.html I'm fine with that as I now understand the syntax

File: themes/src/main/resources/theme/base/email/messages/messages_en.properties [hunk 2]
```diff
@@ -39,18 +39,11 @@ requiredAction.VERIFY_EMAIL=Verify Email
...
+linkExpirationFormatter.timePeriodUnit.minutes={0,choice,0#minutes|1#minute|1<minutes}
+#for languages which have more unit plural forms depending on the value (eg. Czech and other Slavic langs) you can override unit text for some other values like described in the Java choice format: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/text/ChoiceFormat.html
```
Comment (stianst) [inline themes/src/main/resources/theme/base/email/messages/messages_en.properties:44]: I'd link to https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/text/MessageFormat.html as well, or instead?

Comment (ahus1) [reply themes/src/main/resources/theme/base/email/messages/messages_en.properties]: Docs and link to docs are now updated.

**Comment by ahus1** [comment]
@stianst - this is now ready for another review. Thanks!

File: themes/src/main/resources-community/theme/base/email/messages/messages_fa.properties [hunk 3]
```diff
@@ -39,14 +39,9 @@ requiredAction.VERIFY_EMAIL=تأیید ایمیل
 requiredAction.CONFIGURE_RECOVERY_AUTHN_CODES=کدهای بازیابی را ایجاد کنید
 
 linkExpirationFormatter.timePeriodUnit.seconds=ثانیه
```
Comment (stianst) [inline themes/src/main/resources-community/theme/base/email/messages/messages_fa.properties:41]: Should we add a test to make sure this doesn't break when it's not using a ChoiceFormat?

Comment (ahus1) [reply themes/src/main/resources-community/theme/base/email/messages/messages_fa.properties:41]: Added another test `format_simple_no_choice` which passes green.

**Review by stianst — DISMISSED** [review]

File: themes/src/main/resources-community/theme/base/email/messages/messages_cs.properties [hunk 4]
```diff
@@ -35,26 +35,10 @@ requiredAction.UPDATE_PROFILE=Aktualizace profilu
...
-linkExpirationFormatter.timePeriodUnit.hours.1=hodina
-linkExpirationFormatter.timePeriodUnit.hours.2=hodiny
-linkExpirationFormatter.timePeriodUnit.hours.3=hodiny
-linkExpirationFormatter.timePeriodUnit.hours.4=hodiny
-linkExpirationFormatter.timePeriodUnit.days=dní
-linkExpirationFormatter.timePeriodUnit.days.1=den
-linkExpirationFormatter.timePeriodUnit.days.2=dny
-linkExpirationFormatter.timePeriodUnit.days.3=dny
-linkExpirationFormatter.timePeriodUnit.days.4=dny
+linkExpirationFormatter.timePeriodUnit.seconds={0,choice,0#sekund|1#sekunda|2#sekundy|2<sekund}
```
Comment (hmlnarik) [inline themes/src/main/resources-community/theme/base/email/messages/messages_cs.properties:38]: In Czech, the grammatical cases differ with the number up to 5:

- 1 sekunda
- 2 sekundy
- 3 sekundy
- 4 sekundy
- 5 sekund

The current state leads to incorrect "3 sekund" though.

```suggestion
linkExpirationFormatter.timePeriodUnit.seconds={0,choice,0#sekund|1#sekunda|2#sekundy|3#sekundy|4#sekundy|4<sekund}
```

This applies to other locales as well: `ar`, `pl`, `sk`, and to the other time units too.

**Review by hmlnarik — CHANGES_REQUESTED** [review]
I like this PR, this simplifies the automatic translation. Thank you.

However, four locales (ar, cz, sk, pl) lose their translations. Could you please fix?

File: themes/src/main/resources-community/theme/base/email/messages/messages_cs.properties (hunk 4 above)
Comment (ahus1) [reply themes/src/main/resources-community/theme/base/email/messages/messages_cs.properties]: @hmlnarik - thanks for spotting this and sorry for breaking these languages.

I've pushed the necessary changes. The shorter version is to change it to `<2` to `<4`, see below for the output.

Please give it another review when you have the time. Thanks!

```
jshell> java.text.MessageFormat.format("{0,choice,0#sekund|1#sekunda|2#sekundy|4<sekund}",0)
$14 ==> "sekund"

jshell> java.text.MessageFormat.format("{0,choice,0#sekund|1#sekunda|2#sekundy|4<sekund}",1)
$15 ==> "sekunda"

jshell> java.text.MessageFormat.format("{0,choice,0#sekund|1#sekunda|2#sekundy|4<sekund}",2)
$16 ==> "sekundy"

jshell> java.text.MessageFormat.format("{0,choice,0#sekund|1#sekunda|2#sekundy|4<sekund}",3)
$17 ==> "sekundy"

jshell> java.text.MessageFormat.format("{0,choice,0#sekund|1#sekunda|2#sekundy|4<sekund}",4)
$18 ==> "sekundy"

jshell> java.text.MessageFormat.format("{0,choice,0#sekund|1#sekunda|2#sekundy|4<sekund}",5)
$19 ==> "sekund"
```

**Review by hmlnarik — APPROVED** [review]
Thanks for the changes

**Pull request merged**